### PR TITLE
Add autostop

### PR DESCRIPTION
**Pull request by sm9cc**
Automatically stop movement when aimbot has a target to improve accuracy.

Sorry about last 2 pull requests, Still a little rusty.

https://www.unknowncheats.me/forum/counterstrike-global-offensive/258323-autostop.html

### CODE REVIEW

**Comment by thisisprettyepic** [comment]
Crashes when shooting at target

**Comment by SM9CC** [comment]
> Crashes when shooting at target

Ah, I forgot to initialize the ConVars, will sort it out.

**Comment by thisisprettyepic** [comment]
> > Crashes when shooting at target
> 
> Should be fixed, not sure how I missed that! 👍

Still the same thing, appreciate you though.

**Comment by SM9CC** [comment]
> > > Crashes when shooting at target
> > 
> > 
> > Should be fixed, not sure how I missed that! +1
> 
> Still the same thing, appreciate you though.

I have just tested it and it seems to be working fine for me, are you sure its not something else crashing like - https://github.com/danielkrupinski/Osiris/issues/2412

**Comment by thisisprettyepic** [comment]
> > > > Crashes when shooting at target
> > > 
> > > 
> > > Should be fixed, not sure how I missed that! +1
> > 
> > 
> > Still the same thing, appreciate you though.
> 
> I have just tested it and it seems to be working fine for me, are you sure its not something else crashing like - #2412 ?

I’ll try it again today.

**Comment by thisisprettyepic** [comment]
> > > > Crashes when shooting at target
> > > 
> > > 
> > > Should be fixed, not sure how I missed that! +1
> > 
> > 
> > Still the same thing, appreciate you though.
> 
> I have just tested it and it seems to be working fine for me, are you sure its not something else crashing like - #2412 ?

It works now, my bad.

**Comment by doggobot-owner** [comment]
Crashes when using auto shot

**Comment by ooyax** [comment]
crashing?

**Comment by thisisprettyepic** [comment]
Yup it crashes

**Comment by RyDeem** [comment]
dude update your pull, here its working, take this.
i added so it stops stopping when you cant shoot, eg deagle or reload. idk enjoy :)
https://github.com/RyDeem/Ossyrys/tree/AutostopHelp

**Comment by floydGG** [comment]
image

File: Osiris/Hacks/Aimbot.cpp [hunk 1]
```diff
@@ -238,3 +250,48 @@ void Aimbot::run(UserCmd* cmd) noexcept
...
+    ndir.y = cmd->viewangles.y - ndir.y;
+    ndir = ndir.fromAngle(ndir);
+
+    cmd->forwardmove = ndir.x * wishSpeed;
+    cmd->sidemove = ndir.y * wishSpeed;
+}
+
+void Aimbot::init() noexcept
+{
+    cvars.accelerate = interfaces->cvar->findVar("sv_accelerate");
```
Comment (cristyanul) [inline Osiris/Hacks/Aimbot.cpp:295]: pointer to incomplete class type "Cvar" is not allowed

File: Osiris/Hacks/Aimbot.cpp [hunk 2]
```diff
@@ -238,3 +250,48 @@ void Aimbot::run(UserCmd* cmd) noexcept
...
+    cvars.maxSpeed = interfaces->cvar->findVar("sv_maxspeed");
```
Comment (cristyanul) [inline Osiris/Hacks/Aimbot.cpp:296]: pointer to incomplete class type "Cvar" is not allowed

Comment (Ni1kko) [reply Osiris/Hacks/Aimbot.cpp:296]: #include "../SDK/Cvar.h"

**Pull request closed without merging**